client.go

```diff
@@ -63,12 +107,32 @@ func (req *Request) Execute(method, url string) (*Response, error) {
 	if err != nil {
 		return nil, err
 	}
+	for k, _ := range req.QueryParam {
+		req.QueryParam.Del(k)
+	}
 	resp := &Response{restyResponse}
 	return resp, err
 }
 
+func (req *Request) isValid() bool {
+	re := regexp.MustCompile(":name|:digest|:uuid|//{2,}")
+	matches := re.FindAllString(req.URL, -1)
+	if len(matches) == 0 {
+		return true
+	}
+	return false
+}
+
 func (client *Client) Do(req *Request) (*Response, error) {
-	resp, err := req.Execute(req.Method, req.URL)
+	if !req.isValid() {
+		return nil, fmt.Errorf("client must have a namespace set to make requests")
```

Comment:
lets make this error more generic for now, such as "request is invalid" and revisit later - this might also mean `:digest` or `:uuid` is missing